packages/notifications/push/amplify_push_notifications_pinpoint/example/ios/unit_tests/AmplifyPushNotificationsPluginTest.swift

```diff
@@ -0,0 +1,530 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import XCTest
+import Flutter
+
+@testable import amplify_push_notifications
+
+let fakePointer = AutoreleasingUnsafeMutablePointer<FlutterError?>(bitPattern: 1)!
+let expectedNotification = [
+    "aps": [
+        "alert": ["title": "I'm a PN"],
+        "content-available": 1
+    ] as [String : Any]
+]
+let expectedDeviceToken = "abcd1234ef5678";
+let expectedDeviceTokenData = revertTokenOperation(expectedDeviceToken)
+let expectedErrorLocalizedDescription = NSLocalizedString("register token failed", comment: "mockError")
+var fakeIconBadgeNumber = 0
+var fakeUIApplicationState = UIApplication.State.active
+var hasCalledRegisterForRemoteNotifications = false
+var hasOnNotificationReceivedInBackgroundBeenCalled = false
+var hasNotificationOpenedHandlerBeenCalled = false
+var hasForegroundMessageReceivedHandlerBeenCalled = false
+var hasTokenReceivedHandlerBeenCalled = false
+var hasTokenReceivedErrorHandlerBeenCalled = false
```

Comment:
If you're gonna have globals, they need to be reset after each test

Reply:
Added resetting the values in `setUp`.